rules/Naming/PropertyRenamer/MatchTypePropertyRenamer.php

```diff
@@ -22,6 +25,27 @@ public function rename(PropertyRename $propertyRename): ?Property
             return null;
         }
 
-        return $this->propertyRenamer->rename($propertyRename);
+        if ($propertyRename->isAlreadyExpectedName()) {
+            return null;
+        }
+
+        if ($this->propertyRenameGuard->shouldSkip($propertyRename)) {
+            return null;
+        }
+
+        $onlyPropertyProperty = $propertyRename->getPropertyProperty();
+        $onlyPropertyProperty->name = new VarLikeIdentifier($propertyRename->getExpectedName());
+        $this->renamePropertyFetchesInClass($propertyRename);
+
+        return $propertyRename->getProperty();
+    }
+
+    private function renamePropertyFetchesInClass(PropertyRename $propertyRename): void
+    {
+        $this->propertyFetchRenamer->renamePropertyFetchesInClass(
+            $propertyRename->getClassLike(),
+            $propertyRename->getCurrentName(),
+            $propertyRename->getExpectedName()
+        );
```

Comment:
Wouldn't it be better to pass `$propertyRename` object? Since all 3 values comes from it, what's the point of splitting it here?